packages/app/src/app/pages/Dashboard/Content/routes/Settings/TeamSettings/WorkspaceSettings.tsx

```diff
@@ -112,10 +116,34 @@ export const WorkspaceSettings = () => {
     });
   };
 
+  const handleTeamNameChange = event => {
+    const { value } = event.target;
+
+    // Get the input and remove any whitespace from both ends.
+    const trimmedName = value?.trim() ?? '';
+
+    // Validate if the name input is filled with whitespaces.
+    if (!trimmedName) {
+      event.target.setCustomValidity('Team name is required.');
+    } else if (teams.find(t => t.name === trimmedName)) {
+      event.target.setCustomValidity(
+        'Name already taken, please choose a new name.'
+      );
+    } else {
+      event.target.setCustomValidity('');
+    }
+  };
+
   const onSubmit = async event => {
     event.preventDefault();
-    const name = event.target.name.value;
-    const description = event.target.description.value;
+
+    const name = event.target.name.value?.trim();
+    const description = event.target.description.value?.trim();
+
+    if (!name) {
+      return;
+    }
+
     setLoading(true);
     try {
```

Comment:
Should we also add a `catch`?

Reply:
@tristandubbeld I don't think so,`setTeamInfo` catches errors internally and dispatches a notification when they occur. I added this info as a comment to the code.